python/pyarrow/_dataset.pyx

```diff
@@ -508,12 +508,13 @@ cdef class Dataset(_Weakrefable):
 
 
 cdef class InMemoryDataset(Dataset):
-    """A Dataset wrapping in-memory data.
+    """
+    A Dataset wrapping in-memory data.
 
     Parameters
     ----------
-    source
-        The data for this dataset. Can be a RecordBatch, Table, list of
+    source : The data for this dataset.
+        Can be a RecordBatch, Table, list of
```

Comment:
```suggestion
    source : 
        The data for this dataset. Can be a RecordBatch, Table, list of
```

If some content for the type description is needed, I would add something like "table-like"